src/app/parcels/batch/BatchDataRow.ts

```diff
@@ -0,0 +1,377 @@
+    shippingMethod: string;
+    collectionInfo: CollectionInfo | null;
+}
+
+interface OverrideData {
+    client: OverrideClient;
+    parcel: OverrideParcel;
+}
+
+interface BatchEditData {
+    client: BatchClient;
+    clientReadOnly: boolean;
+    parcel: BatchParcel | null;
+}
+interface OverrideDataRow {
+    data: OverrideData;
+}
+
+interface BatchDataRow {
+    id: number;
+    clientId: string | null;
+    data: BatchEditData | null;
+}
+
+export interface BatchTableDataState {
+    overrideDataRow: OverrideDataRow;
+    batchDataRows: BatchDataRow[];
+}
+
+export interface BatchActionType {
+    type: "update_cell" | "add_row" | "delete_row" | "override_column" | "use_existing_client";
+    payload?: BatchActionPayload;
+}
+
+interface BatchActionPayload {
+    rowId?: number;
+    newRow?: BatchDataRow;
+    newOverrideRow?: OverrideDataRow;
+    existingClientId?: string;
+}
+
+const getOverridenFields = (allFields: OverrideClient | OverrideParcel): string[] => {
+    return Object.entries(allFields)
+        .filter(([_, value]) => value)
+        .reduce((acc, [key, _]) => {
+            return [...acc, key];
+        }, [] as string[]);
+};
+
+function createBooleanGroupFromStrings(strings: string[] | null): BooleanGroup {
+    const result: BooleanGroup = {};
+    if (strings) {
+        strings.forEach((str) => {
+            result[str] = true;
+        });
+    }
+    return result;
+}
+
+const getNappySize = (info: string | null): string | null => {
+    if (info) {
+        const match = info.match(/Nappy Size:\s*(\d+)/);
+        if (match) {
+            return match[1];
+        }
+    }
+    return null;
+};
+
+const getChildrenAndAdults = async (
+    familyId: string
+): Promise<{ adults: Person[]; children: Person[] }> => {
+    const { data, error } = await getFamilySupabaseCall(familyId);
+
+    const adults: Person[] = [];
+    const children: Person[] = [];
+
+    const currentYear = new Date().getFullYear();
+    const currentMonth = new Date().getMonth();
+
+    if (error) {
+        logErrorReturnLogId("Error with fetch: family_members", { error: error });
+        return { adults: [], children: [] };
+    } else if (data) {
+        data.forEach((person) => {
+            const formattedPerson: Person = {
+                gender: person.gender,
+                birthYear: person.birth_year,
+                birthMonth: person.birth_month,
+            };
+            if (!formattedPerson.birthMonth) {
+                currentYear - formattedPerson.birthYear >= 16
+                    ? adults.push(formattedPerson)
+                    : children.push(formattedPerson);
+            } else {
+                currentYear - formattedPerson.birthYear > 16 ||
+                (currentYear - formattedPerson.birthYear === 16 &&
+                    currentMonth >= formattedPerson.birthMonth)
+                    ? adults.push(formattedPerson)
+                    : children.push(formattedPerson);
```

Comment:
This if-else seems a bit messy. Not sure if there is a better way but maybe give it a look over. You do the same check (year-birthyear > 17) first in both cases, can you take that out of the if statement at all?

Reply:
Do you think this is cleaner? 
```
const yearsDiff = currentYear - formattedPerson.birthYear
            if( yearsDiff > 16) {
                adults.push(formattedPerson);
            } else {
                !formattedPerson.birthMonth ?
                yearsDiff == 16 ? adults.push(formattedPerson) : children.push(formattedPerson)
                : (yearsDiff == 16 && formattedPerson.birthMonth <= currentMonth) ? adults.push(formattedPerson) : children.push(formattedPerson);
```

Reply:
Better, what about including a check for years diff being equal to and then less than 16? (if, else if, else) Would that work?